module/modmanager/manager.go

```diff
@@ -400,14 +405,21 @@ func (mgr *Manager) startModule(ctx context.Context, mod *module) error {
 		return errors.WithMessage(err, "error while starting module "+mod.cfg.Name)
 	}
 
+	// Does a gRPC dial. Sets up a SharedConn with a PeerConnection that is not yet connected.
 	if err := mod.dial(); err != nil {
 		return errors.WithMessage(err, "error while dialing module "+mod.cfg.Name)
 	}
 
+	// Sends a ReadyRequest and waits on a ReadyResponse. The PeerConnection will async connect
+	// after this, so long as the module supports it.
 	if err := mod.checkReady(ctx, mgr.parentAddr); err != nil {
 		return errors.WithMessage(err, "error while waiting for module to be ready "+mod.cfg.Name)
 	}
 
+	if pc := mod.sharedConn.PeerConn(); mgr.modPeerConnTracker != nil && pc != nil {
```

Comment:
The debugging I was alluding to above was a result of this line coming before `checkReady`. Hence documenting the state changes with the steps in this function.